presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardCleaner.java

```diff
@@ -251,17 +252,15 @@ private void scheduleBackupCleanup()
 
     private void scheduleLocalCleanup()
     {
+        Set<UUID> local = getLocalShards();
+        scheduler.submit(() -> {
+            waitJitterTime();
+            runLocalCleanupAtStartup(local);
+        });
         scheduler.scheduleWithFixedDelay(() -> {
-            try {
-                // jitter to avoid overloading database
-                long interval = this.localCleanerInterval.roundTo(SECONDS);
-                SECONDS.sleep(ThreadLocalRandom.current().nextLong(1, interval));
-            }
-            catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-            }
+            waitJitterTime();
             runLocalCleanup();
-        }, 0, localCleanerInterval.toMillis(), MILLISECONDS);
+        }, localCleanerInterval.toMillis(), localCleanerInterval.toMillis(), MILLISECONDS);
```

Comment:
I think you don't need to delay the start of usual scheduler, because it significantly delays it. Now it'll run after interval + interval (first pass) + interval (actual cleaning). If interval is 1hr then you'll wait 3 hours until regular cleanup deletes something.
On the other hand, since you moved sleeping out of synchronized methods it's fine if scheduled runs first. Next time it'll just find out that some files are already deleted. See if it'll happen according to the current code.